stacks/apps/augury.yml

```diff
@@ -712,63 +711,98 @@ Resources:
 
         // ECS listTask parameters
-        const listParams = {
-          cluster: process.env.ECS_CLUSTER_ARN,
-          desiredStatus: 'RUNNING',
-          maxResults: MAX_FARGATES,
-          startedBy: STARTED_BY,
+        const startedBy = (name) => `${process.env.STARTED_BY}-${name}`;
+        const listParams = (name) => {
+          return {
+            cluster: process.env.ECS_CLUSTER_ARN,
+            desiredStatus: 'RUNNING',
+            maxResults: 10,
+            startedBy: startedBy(name),
+          };
         };
 
         // ECS runTask parameters
-        const runParams = {
-          cluster: process.env.ECS_CLUSTER_ARN,
-          launchType: 'FARGATE',
-          networkConfiguration: {
-            awsvpcConfiguration: {
-              subnets: [process.env.VPC_SUBNET_1, process.env.VPC_SUBNET_2, process.env.VPC_SUBNET_3],
-              assignPublicIp: 'DISABLED',
-              securityGroups: [process.env.SECURITY_GROUP_1, process.env.SECURITY_GROUP_2, process.env.SECURITY_GROUP_3],
+        const runParams = (name) => {
+          return {
+            cluster: process.env.ECS_CLUSTER_ARN,
+            launchType: 'FARGATE',
+            networkConfiguration: {
+              awsvpcConfiguration: {
+                subnets: [
+                  process.env.VPC_SUBNET_1,
+                  process.env.VPC_SUBNET_2,
+                  process.env.VPC_SUBNET_3,
+                ],
+                assignPublicIp: 'DISABLED',
+                securityGroups: [
+                  process.env.SECURITY_GROUP_1,
+                  process.env.SECURITY_GROUP_2,
+                  process.env.SECURITY_GROUP_3,
+                ],
+              },
+            },
+            overrides: {
+              containerOverrides: [
+                {
+                  name: 'augury-workslow',
+                  environment: [
+                    {
+                      name: 'QUEUE',
+                      value: name,
+                    },
+                  ],
+                },
+              ],
             },
-          },
-          platformVersion: '1.4.0',
-          startedBy: STARTED_BY,
-          taskDefinition: process.env.ECS_TASK_DEFINITION_ARN,
-          tags: [
-            { key: "prx:meta:tagging-version", value: "2021-04-07" },
-            { key: "prx:cloudformation:stack-name", value: process.env.STACK_NAME },
-            { key: "prx:cloudformation:stack-id", value: process.env.STACK_ID },
-            { key: "prx:cloudformation:root-stack-name", value: process.env.ROOT_STACK_NAME },
-            { key: "prx:cloudformation:root-stack-id", value: process.env.ROOT_STACK_ID },
-            { key: "prx:ops:environment", value: process.env.ENVIRONMENT_TYPE },
-            { key: "prx:dev:family", value: "Dovetail" },
-            { key: "prx:dev:application", value: "Augury" },
-          ],
+            platformVersion: '1.4.0',
+            startedBy: startedBy(name),
```

Comment:
Since we append the queue name to the `startedBy` for all tasks - we can also list running tasks by queue name, and control concurrency.